projects/microphysics/argo/prog-run-and-eval.yaml

```diff
@@ -0,0 +1,163 @@
+          - name: config
+            value:  "{{workflow.parameters.config}}"
+          - name: tf_model
+            value: "{{workflow.parameters.tf_model}}"
+          - name: on_off_flag
+            value: "{{workflow.parameters.on_off_flag}}"
+    - - name: piggyback-diags
+        template: piggyback-diags
+        when: "{{workflow.parameters.do_piggy}} == true"
+        arguments:
+          parameters:
+          - name: tag
+            value: "{{workflow.parameters.tag}}"
+      - name: prognostic-diags
+        template: prognostic-diags
+        when: "'{{workflow.parameters.on_off_flag}}' == '--online'"
+        arguments:
+          parameters:
+          - name: tag
+            value: "{{workflow.parameters.tag}}"
+          - name: baseline_tag
+            value: "{{workflow.parameters.baseline_tag}}"
+  - name: run-model
+    inputs:
+      parameters:
+      - name: tag
+      - name: config
+      - name: tf_model
+      - name: on_off_flag
+    tolerations:
+    - key: "dedicated"
+      operator: "Equal"
+      value: "med-sim-pool"
+      effect: "NoSchedule"
+    container:
+      image: "us.gcr.io/vcm-ml/prognostic_run:811d4a88b8b8f29d727f2d61f46f96b770c8ab47"
+      imagePullPolicy: Always
+      workingDir: "/fv3net/projects/microphysics"
+      resources:
+        requests:
+          memory: "6Gi"
+          cpu: "7500m"
+        limits:
+          memory: "8Gi"
+          cpu: "7500m"
+      envFrom:
+      - secretRef:
+          name: wandb-andrep
+      env:
+      - name: GOOGLE_APPLICATION_CREDENTIALS 
+        value: /secret/gcp-credentials/key.json
+      volumeMounts:
+      - mountPath: /secret/gcp-credentials
+        name: gcp-key-secret
+      command: ["bash", "-c", "-x", "-e"]
+      args: 
+      - |
+        WANDB_RUN_GROUP="{{inputs.parameters.tag}}"
+        echo "{{inputs.parameters.config}}" > fv3config.yaml
+        python3 scripts/prognostic_run.py \
+          --tag "{{inputs.parameters.tag}}" \
+          --config-path fv3config.yaml \
+          --model "{{inputs.parameters.tf_model}}" \
+          "{{inputs.parameters.on_off_flag}}"
+  - name: piggyback-diags
+    inputs:
+      parameters:
+      - name: tag
+    tolerations:
+    - key: "dedicated"
+      operator: "Equal"
+      value: "med-sim-pool"
+      effect: "NoSchedule"
+    container:
+      image: us.gcr.io/vcm-ml/fv3net:811d4a88b8b8f29d727f2d61f46f96b770c8ab47
+      imagePullPolicy: Always
+      resources:
+        requests:
+          memory: "6Gi"
+          cpu: "3000m"
+        limits:
+          memory: "15Gi"
+          cpu: "3000m"
+      envFrom:
+      - secretRef:
+          name: wandb-andrep
+      env:
+      - name: GOOGLE_APPLICATION_CREDENTIALS 
+        value: /secret/gcp-credentials/key.json
+      volumeMounts:
+      - mountPath: /secret/gcp-credentials
+        name: gcp-key-secret
+      command: ["bash", "-c", "-x", "-e"]
+      args: 
+      - |
+        cd projects/microphysics
```

Comment:
You can better leverage k8s features here. Our use of `bash  -c` is kind of an unsafe workaround. e.g. using `{{inputs.paramters.tag}}`  below could be used to run an arbitrary command/expose a secret.  

In some cases this is necessary, but not here since this command requires no config yaml.

Use `workingDir` to change the working directory and `env` to setup environment variables, and `args` to specify the command to run. I'm not sure why the `LS` variable is required.
```
workingDir: /fv3net/projects/microphysics
env:
  - name: WANDB_RUN_GROUP
    value: "{{inputs.parameter.tag}}"
args:
  - python3
  - scripts/piggy-back.py
  - "{{input.parameters.tag}}"
```


Note that I also specify the WANDB_RUN_GROUP here. This is important for grouping these outputs in the console.

Reply:
Very good point.  `LS` is a forgotten leftover from doing some debugging so I'll go ahead and remove that.